gobblin-metrics/src/main/java/gobblin/metrics/KafkaReporter.java

```diff
@@ -0,0 +1,421 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metered;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+import kafka.javaapi.producer.Producer;
+import kafka.producer.KeyedMessage;
+import kafka.producer.ProducerConfig;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Kafka reporter for Codahale metrics.
+ *
+ * @author ibuenros
+ */
+public class KafkaReporter extends ScheduledReporter {
+
+  private ProducerConfig _config;
+  private Producer<String, byte[]> _producer;
+  private String _topic;
+  private int exceptionCount;
+
+  public Set<String> _tags;
+  public String _host;
+  public String _env;
+
+  ObjectMapper mapper = new ObjectMapper();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaReporter.class);
+
+  /**
+   * Returns a new {@link gobblin.metrics.KafkaReporter.Builder} for {@link gobblin.metrics.KafkaReporter}
+   *
+   * @param registry the registry to report
+   * @return KafkaReporter builder
+   */
+  public static Builder forRegistry(MetricRegistry registry) {
+    return new BuilderImpl(registry);
+  }
+
+  private static class BuilderImpl extends Builder<BuilderImpl> {
+    public BuilderImpl(MetricRegistry registry) {
+      super(registry);
+    }
+
+    @Override
+    protected BuilderImpl self() {
```

Comment:
I tried just using `this` but couldn't get it working. Perhaps you can take a look?